backend/entityservice/views/auth_checks.py

```diff
@@ -59,6 +59,31 @@ def abort_if_invalid_results_token(resource_id, results_token):
         safe_fail_request(403, message=INVALID_ACCESS_MSG)
 
 
+def abort_if_inconsistent_upload(uses_blocking, clk_json):
+    # check if the following combinations are true
+    # - uses_blocking is False AND 'clks' element in upload JSON
+    # - uses_blocking if True AND 'clknblocks' element in upload JSON
+    # otherwise, return safe_fail_request
```

Comment:
Nitpick: This might as well be a docstring. (code comments like this don't come up IDE tooltips)